Bottom: 7ab13a709c32c18a6a63e3cfe6aa048dca0c2c2a
Top:    fdd492cdb13406bafcf14c37b554b2de76a99d5b
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-08-06 16:38:03 -0700

Refresh of dup2

---

diff --git a/src/preload.c b/src/preload.c
index b3cf6ac..085e8b4 100644
--- a/src/preload.c
+++ b/src/preload.c
@@ -922,6 +922,9 @@ int dup2(int oldfd, int newfd)
 	int ret;
 
 	oldfdi = idm_lookup(&idm, oldfd);
+	if (oldfdi && oldfdi->type == fd_fork)
+		fork_passive(oldfd);
+
 	newfdi = idm_lookup(&idm, newfd);
 	if (newfdi) {
 		 /* newfd cannot have been dup'ed directly */
